torch/distributed/_tensor/dispatch.py

```diff
@@ -150,10 +152,11 @@ def _operator_dispatch(
             args_schema.append(arg._spec)
             local_args.append(arg._local_tensor)
             if mesh is not None:
-                if mesh != arg.device_mesh:
-                    raise NotImplementedError(
-                        f"{op_call}: DTensor does not support cross-mesh operation yet!"
-                    )
+                with disable_proxy_modes_tracing(), unset_fake_temporarily():
```

Comment:
I think here we don't need to have these two modes enabled anymore, I changed the hashing behavior so that it does not do tensor compare anymore

Reply:
nice, will take out